### DISCUSSION TITLE
Django like image field in piccolo admin

### DISCUSSION THREAD

**Discussion by sumitsharansatsangi** [piccolo-orm/piccolo_admin#191] [Ideas]
Need Django like Imagefield , so that file can be uploaded from the admin.

Currently, in the piccolo-admin there is no any option for file upload.

**Comment by sinisaos** [comment]
@sumitsharansatsangi The ability to upload files is in the final stages of development. It will be available very soon. Here is the [PR](https://github.com/piccolo-orm/piccolo_admin/pull/189). It will have upload to LocalStorage, S3 compatible storage, single and multifile upload and a lot of other features.

**Reply by sumitsharansatsangi** [reply]
Oh great !!!

**Reply by dantownsend** [reply]
We've released it now: https://piccolo-admin.readthedocs.io/en/latest/media_storage/index.html

**Reply by sumitsharansatsangi** [reply]
Thanks a lot, @dantownsend .

**Reply by sumitsharansatsangi** [reply]
Added support for specifying default ACL like `django-storages` provides for AWS s3.

Currently, its support for private access only.

Also, added support for adding correct metadata (Content-Type). 
eg:- 
For Video with mp4 extension , `Content-Type` will be `video/mp4`, but since it is not provided by `piccolo-admin` , it is currently `binary/octet-stream`

At current it is as follows:-
![Screenshot at 2022-08-15 15-53-11e](https://user-images.githubusercontent.com/45959281/184621445-b7bc0a53-84e6-4215-96b9-aa2792fb7f40.png)

But it should be following:-
![Screenshot at 2022-08-15 15-53-16e](https://user-images.githubusercontent.com/45959281/184621607-ce009a58-2cfc-4493-9f7e-77266c7d0949.png)


Because of incorrect content-type the video is not playing in the browser even with the signed url.
Instead of playing , it is directly downloading.

@dantownsend, @sinisaos ,  Please merge the [pull](https://github.com/piccolo-orm/piccolo_api/pull/171) and make a new release.
